alerts/apps_alerts.libsonnet

```diff
@@ -6,11 +6,35 @@
     prefixedNamespaceSelector: if self.namespaceSelector != null then self.namespaceSelector + ',' else '',
   },
 
+  // TODO move into libs
+  local wrap_rule_for_labels(rule) =
+    // Detect Kind of rule from name unless hidden `kind field is passed in the rule`
+    local kind =
+      if std.objectHas(rule, 'kind') then rule.kind
+      else if std.startsWith(rule.alert, 'KubePod') then 'pod'
+      else if std.startsWith(rule.alert, 'KubeStateful') then 'statefulset'
+      else if std.startsWith(rule.alert, 'KubeDeploy') then 'deployment'
+      else if std.startsWith(rule.alert, 'KubeDaemon') then 'daemonset'
+      else null;
+
+    local labels = { join_labels: $._config['%ss_join_labels' % std.toString(kind)], on_labels: [kind, 'namespace'], metric: 'kube_%s_labels' % std.toString(kind) };
```

Comment:
isn't `kind` already a string?

explicitly calling `std.toString` may hide some type errors; I'd prefer if jsonnet just barfed on us instead of creating a join label called `{"foo":"bar"}_join_label` or `null_join_label`

Reply:
it is not a string when `else null` in line 18. 

maybe i can just use empty string

Reply:
but when rule is null this expression doesn't get evaluated.

jsonnet expressions are lazily evaluated, so when you return `rule` if kind is `null`, you never access `labels` and thus this formatting expression never gets to see `null`